File: Pratica 4 - Estrutura de Dados SImples/java_logic/Exemplo3.java
import java.util.Scanner;

public class Exemplo3 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        //Usamos 'String[]' para arrays de Strings de tamanho fixo
        String[] matriz_A = new String[10];
        
        System.out.println("Digite 10 nomes:");
        for (int i = 0; i < 10; i++) {
            System.out.print("Digite o " + (i + 1) + "o nome: ");
            matriz_A[i] = scanner.nextLine();
        }

        //Apresentar os nomes
        System.out.println("\n--- Nomes Inseridos ---");
        for (String nome : matriz_A) {
            System.out.println(nome);
        }
    }
}

File: Pratica 4 - Estrutura de Dados SImples/cpp_logic/exemplo3.cpp
#include <iostream>
#include <vector>
#include <string>

int main(){
    //Usamos o 'std::vector' para uma lista dinâmica de strings
    std::vector<std::string> matriz_A(10);

   std::cout << "--- Digite 10 nomes ---:" << std::endl;
   for (int i = 0; i < 10; i++) {
       std::cout << "Digite o " << i + 1 << "o nome: ";
       std::getline(std::cin, matriz_A[i]);
   }

   //Apresentar os nomes
   std::cout << "\n--- Nomes Inseridos ---" << std::endl;
   for (const auto& nome : matriz_A) {
       std::cout << nome << std::endl;
   }
   
    return 0;
}